src/app/parser/open-api/param-types.ts

```diff
@@ -275,5 +283,203 @@ export function getEmptySchema(): Schema {
     required: undefined,
     description: undefined,
     _$rendered: undefined,
+    _$requiresRelaxedTypeTag: undefined,
   };
 }
+
+/**
+ * Function to determine if this scalar type schema requires
+ * relaxed type tag
+ *
+ * More Reading: https://github.com/hasura/ndc-nodejs-lambda?tab=readme-ov-file#relaxed-types
+ *
+ * @param schema
+ */
+export function isRelaxedTypeTagRequiredForScalarTypeSchema(
+  schema: SchemaTypeScalar,
+): boolean {
+  return (
+    (scalarSchemaIsObject(schema) || (schema.enum && schema.enum.length > 1)) ??
+    false
+  );
+}
+
+/**
+ * Function to determine if this object type schema requires
+ * relaxed type tag
+ *
+ * More Reading: https://github.com/hasura/ndc-nodejs-lambda?tab=readme-ov-file#relaxed-types
+ *
+ * ### WARNING
+ * This function should be called after all the children/properties of the schema
+ * have been processed. This function *DOES NOT* process the tag requirements for the
+ * schema's children/properties
+ *
+ * @param schema
+ */
+export function isRelaxedTypeTagRequiredForObjectTypeSchema(
+  schema: SchemaTypeObject,
+): boolean {
+  const children = getScehmaTypeObjectChildrenMap(schema);
```

Comment:
`schema` is spelt incorrectly here.

Reply:
Resolved in [0a1d4ad](https://github.com/hasura/ndc-open-api-lambda/pull/46/commits/0a1d4ad52663de93ae2b00976335e1056cfc54e3)